### DISCUSSION TITLE
How to constrain a datatype to be 4 digit pin

### DISCUSSION THREAD

**Discussion by R-Mull** [supabase/supabase#19991] [Questions]
How can I create a 4-digit pin code?
I want to enable users to select anything from 0000 to 9999.
Cheers

**Comment by GaryAustin1** [comment]
You will probably do something like in the answer here for 8 digits.  https://stackoverflow.com/questions/39431720/constrain-digits-postgresql

**Reply by R-Mull** [reply]
Thanks for the reply Gary, but I seem to be screwing it up.
I've selected int4 as the datatype and used different variations of the the constraint "(pin_code ~ '^[0-9]{4}$')" but none of them seem to work.

<img width="565" alt="image" src="https://github.com/supabase/supabase/assets/154046331/f09bf9bf-6ca4-40ce-b50e-aeaef5b10fbe">
<img width="562" alt="image" src="https://github.com/supabase/supabase/assets/154046331/20fba351-8891-4617-ba0b-ef6514412470">

**Reply by GaryAustin1** [reply]
Deleted my comment below.

**Reply by GaryAustin1** [reply]
I would think you would need a string to use like and regex as you want to preserve leading 0's.  
But as this comment there says:
`You need to enclose the regular expression with a ^ and a $ like so: '^[0-9]{8}$'. But you shouldn't really use a regex here. Just use integer for the type and a mynum BETWEEN 0 AND 99999999 constraint on it`
If you use an int type then you have to use BETWEEN AND you will have to format the output to have leading 0's if you desire them to look like a pin.
